rosbridge_library/src/rosbridge_library/capabilities/defragmentation.py

```diff
@@ -87,16 +87,17 @@ def defragment(self, message):
                 time_diff = now - self.received_fragments[id]["timestamp_last_append"]
                 if (time_diff.total_seconds() > self.fragment_timeout and
                     not self.received_fragments[id]["is_reconstructing"]):
-                    log_msg = "fragment list " + str(id) + " timed out.."
+                    log_msg = ["fragment list " + str(id) + " timed out.."]
 
                     if message["id"] != id:
-                        log_msg += " -> removing it.."
+                        log_msg.append(" -> removing it..")
                         del self.received_fragments[id]
                     else:
-                        log_msg += " -> but we're just about to add fragment #"
-                        log_msg += str(message.get("num")) + " of "
-                        log_msg += str(self.received_fragments[message.get("id")]["total"])
-                        log_msg += " ..keeping the list"
+                        log_msg.append(" -> but we're just about to add fragment #")
+                        log_msg.append(str(message.get("num")) + " of ")
+                        log_msg.append(str(self.received_fragments[message.get("id")]["total"]))
+                        log_msg.append(" ..keeping the list")
+                        log_msg = ''.join(log_msg)
```

Comment:
This join is indented incorrectly - you're only joining in the `else` clause.
To make it harder to forget, you can join right in the log call:
```
self.protocol.log('warning', ''.join(log_msg))
```